sdk/core/azure-core-http-jdk11-httpclient/src/main/java/com/azure/core/http/jdk11/httpclient/Jdk11AsyncHttpClient.java

```diff
@@ -6,57 +6,59 @@
 import com.azure.core.http.HttpClient;
 import com.azure.core.http.HttpHeader;
 import com.azure.core.http.HttpHeaders;
-import com.azure.core.http.HttpMethod;
 import com.azure.core.http.HttpRequest;
 import com.azure.core.http.HttpResponse;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.FluxUtil;
 import reactor.adapter.JdkFlowAdapter;
 import reactor.core.Exceptions;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
-import java.io.IOException;
-import java.io.InputStream;
 import java.net.URISyntaxException;
 import java.nio.ByteBuffer;
 import java.nio.charset.Charset;
-import java.util.concurrent.CompletableFuture;
+import java.util.Collections;
+import java.util.List;
 
 import java.net.http.HttpRequest.BodyPublisher;
+import java.util.Set;
+import java.util.TreeSet;
 import java.util.concurrent.Flow;
 
 import static java.net.http.HttpResponse.BodyHandlers.*;
 import static java.net.http.HttpRequest.BodyPublishers.*;
-import static java.net.http.HttpResponse.BodyHandlers.ofInputStream;
 
 /**
  * HttpClient implementation for the JDK 11 HttpClient.
  */
 class Jdk11AsyncHttpClient implements HttpClient {
     private final java.net.http.HttpClient jdk11HttpClient;
 
-    private static final Mono<BodyPublisher> EMPTY_BODY_PUBLISHER_MONO = Mono.just(noBody());
+    private static final Set<String> JDK11_RESTRICTED_HEADERS;
+    static {
+        TreeSet<String> treeSet = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
+        treeSet.addAll(Set.of("connection",
+            "content-length",
+            "date", // BUG: Java11: https://bugs.openjdk.java.net/browse/JDK-8213189 (Fixed in JDK12)
```

Comment:
@JonathanGiles I think this bug in JDK11 HttpClient may be a blocker for this HttpClient adoption. "Date" is a necessary header for many of the track2 libs, something we need to look into.

Reply:
In that case we would only support this on JDK 12 and later, which is unfortunate.

Reply:
added necessary validation to log error in case this client is ever used with java_version <= 11.